### PR TITLE
feat(DRIVERS-1108): add service host to mech props

### PR DESCRIPTION
**Pull request by durran**
Updates the auth spec to allow drivers to allow a `SERVICE_HOST` property in their auth mechanism properties, which replaces the default host in the service principal name to the provided attribute.

Node impl: https://github.com/mongodb/node-mongodb-native/pull/3123

### CODE REVIEW

**Review by JamesKovacs — APPROVED** [review]
LGTM.

Discussed with Durran whether we can just incorporate `SERVICE_HOST` into the `SERVICE_NAME` as `authMechanismProperties=SERVICE_NAME:other%40example.com`. The .NET/C# driver correctly parses the service name as `other@example.com`, but we have other code that appends the `hostname` automatically. This would have to be altered and Windows/Linux differences in `serviceName/hostname@realm` on Windows versus `serviceName@hostname` on Linux would have to be dealt with. Easier to override the `hostname` with `SERVICE_HOST` than trying to parse `SERVICE_NAME`.

In summary let's add `SERVICE_HOST` to `authMechanismProperties` as proposed.
